docs/design/fabric-v2+/fpc-lifecycle-v2.puml

```diff
@@ -15,124 +15,149 @@ title Foot Box removed
+
+    The current bookkeeping ops expect the approve and
+    commit ops to run on peers where the FPC chaincode
+    has been previously installed.
+    end note
+
+    Peer_CLI -> Peer_CLI : retrieve cc_language from package
+    Peer_CLI -> Peer_CLI : if cc_language==fpc\n\tthen store packageId
+    return packageId
+end
+
+ref over Admin, Peer_CLI, Peer2
+    Admin installs chaincode on Peer2
+end
+
+group approveformyorg
+    Admin -> Peer_CLI++ : lifecycle\nchaincode approveformyorg
+    note right
+    **Chaincode Definition for FPC Chaincode** including
     Name: cc_name,
     Version: **mrenclave**,
-    Sequence: sequence_number, Endorsement Policy: (2of3),
-    Plugins: <ecc validation plugin>
-    NOTE: the endorsement policy specifies how many enclaves are expected to run the chaincode.
-    In the initial version of FPC, a specific FPC chaincode will be implemented to run on
-    a "designated" peer. Hence, initially, the endorsement policy will require a single endorser.
-    FPC is expected to support multiple endorsers in the future. To have an idea of how FPC will
-    enalve multiple endorsers, please check out the diagram describing
-    how secret/private keys are expected to be distributed among endorsing chaincode enclaves.
+    Sequence: sequence_number,
+    Endorsement Policy: (2of3),
+    Plugins: ecc-vscc (FPC validation plugin)
+
+    NOTE: the admin approves the chaincode definition
+    including the **mrenclave** that was computed during
+    code inspection.
+
+    NOTE: the endorsement policy specifies how many
+    enclaves are expected to run the chaincode. The MVP
+    implementation will support a "designated" endorsing
+    peer only.
+    Hence, initially, the endorsement policy will require
+    a single endorser. FPC is expected to support
+    multiple endorsers in the future. To have an
+    idea of how FPC will enclave multiple endorsers,
+    please check out key distribution diagram.
+
     NOTE: FPC does not support custom user plugins.
-end note
-Peer1_CLI -> Orderer ++: approve transaction
-return transaction committed /' Peer1_CLI -> Orderer '/
-return /' FPC_Peer1_CLI_Wrapper -> Peer1_CLI '/
-note over FPC_Peer1_CLI_Wrapper
-    Bookkeping operations.
-    Note: packageId is an input parameter of approveformyorg
-end note
-FPC_Peer1_CLI_Wrapper -> FPC_Peer1_CLI_Wrapper : if packageId was stored (and hence it is an FPC chaincode) then store mapping packageid to <cc_name, cc_version>
-return /' Admin -> FPC_Peer1_CLI_Wrapper '/
-end /' group approve '/
-
-group other orgs approve
-Channel -> Channel : approvals from other parties
-note over Channel
-    Other participants must approve the chaincode definition,
-    and commit the corresponding transaction,
-    according to the LifecycleEndorsement policy.
-    So the party here has to loop to check for the approvals.
-end note
-end group 'other orgs approve'
+    end note
+
+    Peer_CLI -> Orderer++ : approve transaction
```

Comment:
I think the peer is actually also (first) directly contacted by the cli.  However, i think for simplicity removing that flow here (and commit where i think the same applies) seems fine ...